libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -309,6 +309,8 @@ public void setLocationEngine(@NonNull LocationEngine locationEngine) {
     // Notify service to get new location engine.
     if (isServiceAvailable()) {
       navigationService.acquireLocationEngine();
+      // Update location engine name in the session state
+      sessionState.toBuilder().locationEngineName(locationEngine.getClass().getSimpleName());
```

Comment:
Be careful because this not always executed.
BTW, we're not doing anything here.
Even if we do `sessionState = sessionState.toBuilder().locationEngineName(locationEngine.getClass().getSimpleName()).build();` it won't work because 👉 https://github.com/mapbox/mapbox-navigation-android/pull/382/files#r144950030 and because we'll need to recreate the whole `sessionState` adding each value to the different parameters (`toBuilder()` doesn't return the current values from the object 😞).